engine/gamesys/src/gamesys/test/resource/script_buffer.script

```diff
@@ -75,10 +75,30 @@ function test_create_ownership_data(self)
     for i=1,3 do
         assert(buffer_stream[i] == res_buffer_stream[i])
     end
+
+    resource.release(res_buffer)
+    resource.release(res_buffer)
+end
+
+function buffer_to_handle(buf)
+    local res_buf_str = tostring(buf)
+    local res_buf_handle = string.match(res_buf_str, "handle = (%d+)")
+    return tonumber(res_buf_handle)
+end
+
+function test_create_set(self)
+    local buf        = buffer.create(3, data_stream_desc)
+    local buf_bigger = buffer.create(6, data_stream_desc)
+    local res_buffer = resource.create_buffer("/my_buffer.bufferc", { buffer = buf })
+    resource.set_buffer("/my_buffer.bufferc", buf_bigger, { transfer_ownership=true })
```

Comment:
I'm not 100% sure I follow what you are after, but maybe the following scenario?

```
local path          = "/my_buffer.bufferc"
local buffer_res    = make_buffer_res(path)
local buffer_handle = make_buffer(0.25, 1)

local buffer_at_path = resource.get_buffer(path)
local stream_at_path = buffer.get_stream(buffer_at_path, hash("position"))

-- the 'old' buffer (buffer_at_path) will get removed now in favor of the incoming buffer
-- using the buffer_at_path later on will still be fine since the lua buffer handle
-- has been updated with the new handle, but the stream reference we got ealier (stream_at_path)
-- will contain a reference to the old handle, so modifying it wont work unless it is updated.
resource.set_buffer(path, buffer_handle, { transfer_ownership = true })
resource.set_buffer(path, buffer_handle, { transfer_ownership = true })

-- this wont work!
stream_at_path[1] = 128
-- but this will
stream_at_path = buffer.get_stream(buffer_at_path, hash("position"))
stream_at_path[1] = 128
```

The 'this wont work' line will cause a lua error, but not a full crash. There might be some way of causing a crash, perhaps by transferring the same buffer between different resources, but so far I haven't figured out how 🤔